### PR TITLE
General signing

### PR DESCRIPTION
**Pull request by alex-miller-0**
Adds functionality for signing any message on a secp256k1 or ed25519 curve.

Corresponds to firmware feature: https://github.com/GridPlus/k8x_firmware_production/pull/2339

### CODE REVIEW

**Comment by alex-miller-0** [comment]
This line needs to be changed before this can be reviewed/used: https://github.com/GridPlus/gridplus-sdk/pull/250/commits/37de5179282ad921283ec9c3bc75552e31c1eb8f#diff-8fa4b52909f895e8cda060d2035234e0a42ca2c7d3f8f8de1b35a056537bf199R340

File: test/testWalletJobs.ts [hunk 1]
```diff
@@ -528,7 +528,7 @@ describe('signTx', () => {
       payload: null,
     };
   });
-
+/*
```
Comment (alex-miller-0) [inline test/testWalletJobs.ts:531]: Cruft

**Review by douglance — APPROVED** [review]
Thanks for refactoring how functions are exported.

File: src/client.ts [hunk 2]
```diff
@@ -348,9 +353,17 @@ export class Client {
       schema = signingSchema.EXTRA_DATA;
     } else {
       try {
-        req = signReqResolver[currency](data);
+        if (currency === 'ETH') {
+          req = ethereum.buildEthereumTxRequest(data);
+        } else if (currency == 'ETH_MSG') {
+          req = ethereum.buildEthereumMsgRequest(data);
+        } else if (currency == 'BTC') {
```
[the remark this answers is not in the document]
Comment (alex-miller-0) [reply src/client.ts]: [348affb](https://github.com/GridPlus/gridplus-sdk/pull/250/commits/348affb4ef7017d76f2b89c3f80e7083b36c9345)

File: test/testWalletJobs.ts (hunk 1 above)
Comment (alex-miller-0) [reply test/testWalletJobs.ts]: [348affb](https://github.com/GridPlus/gridplus-sdk/pull/250/commits/348affb4ef7017d76f2b89c3f80e7083b36c9345)

**Pull request merged**